x-pack/plugin/snapshot-repo-test-kit/qa/s3/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/S3SnapshotRepoTestKitIT.java

```diff
@@ -61,6 +61,11 @@ protected Settings repositorySettings() {
         final String basePath = System.getProperty("test.s3.base_path");
         assertThat(basePath, not(blankOrNullString()));
 
-        return Settings.builder().put("client", "repo_test_kit").put("bucket", bucket).put("base_path", basePath).build();
+        return Settings.builder()
+            .put("client", "repo_test_kit")
+            .put("bucket", bucket)
+            .put("base_path", basePath)
+            .put("deletion_batch_size", between(1, 1000))
```

Comment:
I took the easy way to piggyback on the existing repo analysis test. Please let me know if this works for you. Thanks!

Reply:
:+1: should be fine, but could you just confirm that this exposes the bug?

Reply:
Yes I tested locally with real s3 by setting the properties listed [here](https://github.com/elastic/elasticsearch/blob/5a6f8cdb97a7e2493e8bebf04bb3d9c9a304e143/x-pack/plugin/snapshot-repo-test-kit/qa/s3/build.gradle#L43-L48). I can confirm the test can fail without the fix. 

A few notes:
1. It does not fail with the fixture. I wonder whether it has something to do with the fixture running on localhost.
2. When the size is 1, it will fail at repo verification without even start running the test. 
3. I expected it to fail anytime the deletion is done inside the lambda but that is not always the case. I guess maybe if there is existing connection to perform the deletion request, it does not trigger the permission check. I did not look into it deeply since it feels sufficient to me that it does fail sometimes without the fix.